File: src/array/2225.cpp
class Solution
{
public:
  vector<vector<int>> findWinners(vector<vector<int>> &matches)
  {
    unordered_map<int, int> map; // 当前用户输给了多少人
    vector<vector<int>> result(2);

    for (auto item : matches)
    {
      if (!map.count((item[0])))
      {
        map[item[0]] = 0;
      }
      map[item[1]]++;
    }
    for (const auto &[key, value] : map)
    {
      if (value < 2)
      {
        result[value].push_back(key);
      }
    }
    sort(result[0].begin(), result[0].end());
    sort(result[1].begin(), result[1].end());

    return result;
  }
};

File: src/array/2225.js
/**
 * @param {number[][]} matches
 * @return {number[][]}
 */
var findWinners = function (matches) {
  const map = new Map();
  const result = [[], []];

  matches.forEach(([winner, loser]) => {
    map.set(loser, (map.get(loser) || 0) + 1);
    map.set(winner, map.get(winner) || 0);
  });

  for (const [key, value] of map) {
    if (value < 2) {
      result[value].push(key);
    }
  }
  result[0].sort((x, y) => (x <= y ? -1 : 1));
  result[1].sort((x, y) => (x <= y ? -1 : 1));

  return result;
};

console.log(
  findWinners([
    [1, 3],
    [2, 3],
    [3, 6],
    [5, 6],
    [5, 7],
    [4, 5],
    [4, 8],
    [4, 9],
    [10, 4],
    [10, 9],
  ])
);
